acceptance/tests/facts/validate_file_system_size_bytes.rb

```diff
@@ -0,0 +1,39 @@
+# This test verifies that the numbers for file systems sizes are positive from "facter" and "puppet facts"
+# This is a regression test for FACT-1578
+test_name "C100110: verify that file system sizes are positive" do
+  tag 'risk:high'
+
+  require 'json'
+
+  agents.each do |agent|
+    step("verify that facter returns positive numbers for the mount points byte fields") do
+
+      on(agent, facter("--json")) do |facter_output|
+        facter_results = JSON.parse(facter_output.stdout)
+        facter_results['mountpoints'].each_key do |mount_key|
```

Comment:
FYI, you can use the `expect_failure` helper to allow you to merge the test prior to having the ticket fixed. Once the issue is addressed, it will go red and can be removed.

Example
```
['available_bytes',  'size_bytes', 'used_bytes'].each do |subkey|
  if agent['platform'] =~ /solaris/ then
    expect_failure('expected to fail due to FACT-1578') do
     assert_operator(facter_results['mountpoints'][mount_key][subkey], :>=, 0,
                     "Expected the #{subkey} from facter to be positive for #{mount_key}")
    end
  else
     assert_operator(facter_results['mountpoints'][mount_key][subkey], :>=, 0,
                     "Expected the #{subkey} from facter to be positive for #{mount_key}")
  end
end
```